vumi/components/message_store.py

```diff
@@ -76,18 +100,24 @@ class OutboundMessage(Model):
     # Extra fields for compound indexes
     batches_with_timestamps = ListOf(Unicode(), index=True)
     batches_with_addresses = ListOf(Unicode(), index=True)
+    batches_with_addresses_reverse = ListOf(Unicode(), index=True)
 
     def save(self):
         # We override this method to set our index fields before saving.
-        batches_with_timestamps = []
-        batches_with_addresses = []
-        timestamp = format_vumi_date(self.msg['timestamp'])
+        self.batches_with_timestamps = []
+        self.batches_with_addresses = []
+        self.batches_with_addresses_reverse = []
+        timestamp = self.msg['timestamp']
+        if not isinstance(timestamp, basestring):
+            timestamp = format_vumi_date(timestamp)
```

Comment:
I don't think there's any particular reason to. If anything ever needs the other branch of the conditional, tests for that thing will cover it. I'd rather not add the additional runtime to the test suite unless there's enough value to justify it.